internal/envoy/cluster.go

```diff
@@ -43,9 +43,12 @@ func Cluster(s dag.Service) *v2.Cluster {
 
 func httpCluster(service *dag.HTTPService) *v2.Cluster {
 	c := cluster(&service.TCPService)
+
 	switch service.Protocol {
+	case "tls":
+		c.TlsContext = UpstreamTLSContext(service.Protocol)
```

Comment:
Then we can make this method clearer but removing the fallthrough.

case "tls":
     c.TlsContext = UpstreamTLSContext()
case "h2":
     c.TlsContext = UpstreamTLSContext("h2")
case "h2c":
     c.TlsContext = UpstreamTLSContext("h2")
     c.Http2ProtocolOptions = &core.Http2ProtocolOptions{}
}
```